code/modules/atmospherics/machinery/pipes/layermanifold.dm

```diff
@@ -0,0 +1,133 @@
+/obj/machinery/atmospherics/pipe/layer_manifold
+	name = "pipe-layer manifold"
+	icon = 'icons/obj/atmospherics/pipes/manifold.dmi'
+	icon_state = "manifoldlayer"
+	desc = "A special pipe to bridge pipe layers with."
+	dir = SOUTH
+	initialize_directions = NORTH|SOUTH
+	pipe_flags = PIPING_ALL_LAYER
+	piping_layer = PIPING_LAYER_DEFAULT
+	device_type = 0
+	volume = 260
+	var/list/front_nodes
+	var/list/back_nodes
+
+/obj/machinery/atmospherics/pipe/layer_manifold/setPipingLayer()
+	return ..(PIPING_LAYER_DEFAULT)
+
+/obj/machinery/atmospherics/pipe/layer_manifold/Initialize()
+	front_nodes = list()
+	back_nodes = list()
+	return ..()
+
+/obj/machinery/atmospherics/pipe/layer_manifold/Destroy()
+	nullifyAllNodes()
+	return ..()
+
+/obj/machinery/atmospherics/pipe/layer_manifold/proc/nullifyAllNodes()
+	var/list/obj/machinery/atmospherics/needs_nullifying = get_all_connected_nodes()
+	front_nodes = null
+	back_nodes = null
+	nodes = list()
+	for(var/obj/machinery/atmospherics/A in needs_nullifying)
+		A.disconnect(src)
+		A.build_network()
+
+/obj/machinery/atmospherics/pipe/layer_manifold/proc/get_all_connected_nodes()
+	var/list/obj/machinery/atmospherics/all_connected = list()
+	for(var/obj/machinery/atmospherics/I in front_nodes)
+		all_connected[I] = I
+	for(var/obj/machinery/atmospherics/I in back_nodes)
+		all_connected[I] = I
+	for(var/obj/machinery/atmospherics/I in nodes)
+		all_connected[I] = I
+	var/list/obj/machinery/atmospherics/returnlist = list()
+	for(var/obj/machinery/atmospherics/A in all_connected)
+		returnlist += all_connected[A]
+	return returnlist

```

Comment:
is there some reason this proc can't just be `return front_nodes | back_nodes | nodes`?

Reply:
if there's any duplicates it might run into issues...

Reply:
why would there be duplicates? even still, there shouldn't be trouble if you use the list `|` operator